ssl/tls13_enc.c

```diff
@@ -404,6 +404,9 @@ int tls13_change_cipher_state(SSL *s, int which)
                        SSL_R_BAD_HANDSHAKE_LENGTH);
                 goto err;
             }
+
+            if (sslcipher == NULL && s->psksession != NULL)
```

Comment:
Can we end up with a sslcipher in the "regular session" when a psksession is in use?
(If not, this could replace checking for a cipher suite stored in the psksession at the other two places I noted, but it's unclear that we want to wait until CCS to detect such errors.)